src/components/nodes/LabelUpdaterNode.tsx

```diff
@@ -0,0 +1,110 @@
+import { CloseIcon } from "@chakra-ui/icons";
+import { Box, Input } from "@chakra-ui/react";
+import { FormEvent, useEffect, useRef, useState } from "react";
+import { Handle, Position, useReactFlow } from "reactflow";
+import { Row } from "../../utils/chakra";
+import { modifyFluxNodeLabel, modifyFluxNode } from "../../utils/fluxNode";
+import { FluxNodeData } from "../../utils/types";
+
+export function LabelUpdaterNode({
+  id,
+  data,
+  isConnectable,
+}: {
+  id: string;
+  data: FluxNodeData;
+  isConnectable: boolean;
+}) {
+  const { setNodes } = useReactFlow();
+
+  const [renameLabel, setRenameLabel] = useState(data.label);
+  const renameBlockRef = useRef<HTMLInputElement>(null);
+
+  const cancel = () => {
+    setNodes((nodes) =>
+      modifyFluxNode(nodes, {
+        id,
+      })
+    );
+  };
+
+  const renameNode = () => {
+    setNodes((nodes) =>
+      modifyFluxNodeLabel(nodes, {
+        id,
+        label: renameLabel,
+      })
+    );
+  };
+
+  const submit = (e: FormEvent) => {
+    e.preventDefault();
+    renameNode();
+  };
+
+  useEffect(() => {
+    function handleClick(event: any) {
+      if (renameBlockRef.current && !renameBlockRef.current.contains(event.target)) {
+        cancel();
+      }
+    }
+    function handleClose(event: any) {
+      if (event.key === "Escape") cancel();
+    }
+
+    // Bind the event listeners
+    document.addEventListener("mousedown", handleClick);
+    document.addEventListener("keydown", handleClose);
+    return () => {
+      // Unbind the event listeners on clean up
+      document.removeEventListener("mousedown", handleClick);
+      document.removeEventListener("keydown", handleClose);
+    };
+  }, [renameBlockRef]);
```

Comment:
imo lets just kill the mousedown stuff for the sake of keeping things simple

i think its fine that the only way out is either clicking: cancel, typing escape, or hitting return

can also simplify the escape logic by using `useHotkeys` like we do throughout the app i think?

Reply:
I felt clicking outside the box was needed as otherwise if user forgets about clicking return/esc and clicks around, the node would stay custom. Nothing should break, just thought it would feel weird. 

Will look into using hotkeys also

Reply:
hmm yeah ok thats fair, just played around w/ it and could see being unnecessarily confusing — can leave in then